src/_pytest/config/__init__.py

```diff
@@ -739,9 +741,19 @@ def consider_pluginarg(self, arg: str) -> None:
                     del self._name2plugin["pytest_" + name]
             self.import_plugin(arg, consider_entry_points=True)
 
-    def consider_conftest(self, conftestmodule: types.ModuleType) -> None:
+    def consider_conftest(
+        self, conftestmodule: types.ModuleType, registration_name: Optional[str] = None
+    ) -> None:
         """:meta private:"""
-        self.register(conftestmodule, name=conftestmodule.__file__)
+        if registration_name is None:
+            warnings.warn(
+                "'consider_conftest' was called with registration_name=None, this will become an error in the future'",
+                PytestDeprecationWarning,
+            )
+            # Keep Pytest < 8 previous behaviour
+            registration_name = conftestmodule.__file__
```

Comment:
I read the discussion, but to be honest I don't think we need to deprecate receiving None here.

@RonnyPfannschmidt what issue do you see that could be a problem accepting `None` as `registration_name`, and fallback to the previous behavior, which is what has been used for ages now? 

If we feel strongly that the new parameter is needed, giving that it seems like it would break very little plugins that we know of and the "fix" is trivial, I would rather we just break the API then, but I'm not convinced that we need to, accepting `None` and fallback to the previous behavior seems acceptable to me.

So my preference would be (in order):

1. Use `None` as default for `registration_name` and fallback to the previous behavior.
2. Break the API and make `registration_name` mandatory.

Reply:
Sounds good to me. 👍